--- a/pylsp_black/plugin.py
+++ b/pylsp_black/plugin.py
@@ -128,6 +128,7 @@
         # raised when black produces invalid Python code or formats the file
         # differently on the second pass
         black.parsing.ASTSafetyError,
+        black.parsing.SourceASTParseError,
     ) as e:
         # errors will show on lsp stderr stream
         logger.error("Error formatting with black: %s", e)